Standalone bevy_ecs tests fail to compile

## Bevy version

ed13d9b0e2572c9ce6260dd8f5fd420baf4023a3

## What you did

Attempted to run a single test in `bevy_ecs`.

The exact command run was:

```
cargo test --package bevy_ecs --lib --all-features -- world::error::tests::fixing_panicking_entity_commands --exact --show-output
```

That particular test doesn't exist on `main`: I was attempting to fix a different issue.

## What went wrong

```
error[E0599]: no method named `catch_unwind` found for struct `AssertUnwindSafe<impl Future<Output = _>>` in the current scope
   --> crates/bevy_tasks/src/task_pool.rs:484:22
    |
483 |                   let _result = AssertUnwindSafe(executor.run(tick_forever))
    |  _______________________________-
484 | |                     .catch_unwind()
    | |                     -^^^^^^^^^^^^ method not found in `AssertUnwindSafe<impl Future<Output = _>>`
    | |_____________________|
    |

error[E0599]: no method named `catch_unwind` found for struct `AssertUnwindSafe<{async block@crates/bevy_tasks/src/task_pool.rs:500:36: 500:41}>` in the current scope
   --> crates/bevy_tasks/src/task_pool.rs:505:62
    |
505 |                 let _result = AssertUnwindSafe(tick_forever).catch_unwind().await.is_ok();
    |                                                              ^^^^^^^^^^^^ method not found in `AssertUnwindSafe<{async block@crates/bevy_tasks/src/task_pool.rs:500:36: 500:41}>`

error[E0599]: no method named `catch_unwind` found for struct `AssertUnwindSafe<impl Future<Output = _>>` in the current scope
   --> crates/bevy_tasks/src/task_pool.rs:525:22
    |
524 |                   let _result = AssertUnwindSafe(executor.run(tick_forever))
    |  _______________________________-
525 | |                     .catch_unwind()
    | |                     -^^^^^^^^^^^^ method not found in `AssertUnwindSafe<impl Future<Output = _>>`
    | |_____________________|
    |

error[E0599]: no method named `catch_unwind` found for struct `AssertUnwindSafe<{async block@crates/bevy_tasks/src/task_pool.rs:540:36: 540:41}>` in the current scope
   --> crates/bevy_tasks/src/task_pool.rs:545:62
    |
545 |                 let _result = AssertUnwindSafe(tick_forever).catch_unwind().await.is_ok();
    |                                                              ^^^^^^^^^^^^ method not found in `AssertUnwindSafe<{async block@crates/bevy_tasks/src/task_pool.rs:540:36: 540:41}>`

error[E0599]: no method named `catch_unwind` found for struct `AssertUnwindSafe<Fut>` in the current scope
   --> crates/bevy_tasks/src/task_pool.rs:649:40
    |
649 |             .spawn(AssertUnwindSafe(f).catch_unwind())
    |                                        ^^^^^^^^^^^^ method not found in `AssertUnwindSafe<Fut>`

error[E0599]: no method named `catch_unwind` found for struct `AssertUnwindSafe<Fut>` in the current scope
   --> crates/bevy_tasks/src/task_pool.rs:665:40
    |
665 |             .spawn(AssertUnwindSafe(f).catch_unwind())
    |                                        ^^^^^^^^^^^^ method not found in `AssertUnwindSafe<Fut>`

error[E0599]: no method named `catch_unwind` found for struct `AssertUnwindSafe<Fut>` in the current scope
   --> crates/bevy_tasks/src/task_pool.rs:682:40
    |
682 |             .spawn(AssertUnwindSafe(f).catch_unwind())
    |                                        ^^^^^^^^^^^^ method not found in `AssertUnwindSafe<Fut>`

error[E0432]: unresolved import `__automatic_type_registration_impl`
   --> crates/bevy_reflect/src/lib.rs:820:17
    |
820 |         pub use __automatic_type_registration_impl::*;
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ use of unresolved module or unlinked crate `__automatic_type_registration_impl`
    |
    = help: if you wanted to use a crate named `__automatic_type_registration_impl`, use `cargo add __automatic_type_registration_impl` to add it to your `Cargo.toml`
```


## Additional information

Likely introduced by #15030.

--- a/crates/bevy_asset/src/io/file/sync_file_asset.rs
+++ b/crates/bevy_asset/src/io/file/sync_file_asset.rs
@@ -140,10 +140,10 @@ impl AssetReader for FileAssetReader {
                     f.ok().and_then(|dir_entry| {
                         let path = dir_entry.path();
                         // filter out meta files as they are not considered assets
-                        if let Some(ext) = path.extension().and_then(|e| e.to_str()) {
-                            if ext.eq_ignore_ascii_case("meta") {
-                                return None;
-                            }
+                        if let Some(ext) = path.extension().and_then(|e| e.to_str())
+                            && ext.eq_ignore_ascii_case("meta")
+                        {
+                            return None;
                         }
                         // filter out hidden files. they are not listed by default but are directly targetable
                         if path

--- a/crates/bevy_math/src/bounding/bounded2d/primitive_impls.rs
+++ b/crates/bevy_math/src/bounding/bounded2d/primitive_impls.rs
@@ -4,15 +4,15 @@ use crate::{
     bounding::BoundingVolume,
     ops,
     primitives::{
-        Annulus, Arc2d, Capsule2d, Circle, CircularSector, CircularSegment, ConvexPolygon, Ellipse,
-        Line2d, Plane2d, Rectangle, RegularPolygon, Rhombus, Segment2d, Triangle2d,
+        Annulus, Arc2d, Capsule2d, Circle, CircularSector, CircularSegment, Ellipse, Line2d,
+        Plane2d, Rectangle, RegularPolygon, Rhombus, Segment2d, Triangle2d,
     },
     Dir2, Isometry2d, Mat2, Rot2, Vec2,
 };
 use core::f32::consts::{FRAC_PI_2, PI, TAU};
 
 #[cfg(feature = "alloc")]
-use crate::primitives::{Polygon, Polyline2d};
+use crate::primitives::{ConvexPolygon, Polygon, Polyline2d};
 
 use smallvec::SmallVec;
 

--- a/crates/bevy_math/src/bounding/bounded3d/extrusion.rs
+++ b/crates/bevy_math/src/bounding/bounded3d/extrusion.rs
@@ -126,6 +126,7 @@ impl BoundedExtrusion for Rectangle {
     }
 }
 
+#[cfg(feature = "alloc")]
 impl BoundedExtrusion for Polygon {
     fn extrusion_aabb_3d(&self, half_depth: f32, isometry: impl Into<Isometry3d>) -> Aabb3d {
         let isometry = isometry.into();

--- a/crates/bevy_math/src/primitives/dim2.rs
+++ b/crates/bevy_math/src/primitives/dim2.rs
@@ -1,5 +1,6 @@
 use core::f32::consts::{FRAC_1_SQRT_2, FRAC_PI_2, FRAC_PI_3, PI};
 use derive_more::derive::From;
+#[cfg(feature = "alloc")]
 use thiserror::Error;
 
 use super::{Measured2d, Primitive2d, WindingOrder};
@@ -1996,6 +1997,7 @@ impl ConvexPolygon {
     }
 }
 
+#[cfg(feature = "alloc")]
 impl TryFrom<Polygon> for ConvexPolygon {
     type Error = ConvexPolygonError;
 

--- a/crates/bevy_reflect/src/lib.rs
+++ b/crates/bevy_reflect/src/lib.rs
@@ -817,6 +817,7 @@ pub mod __macro_exports {
             }
         }
 
+        #[cfg(any(feature = "auto_register_static", feature = "auto_register_inventory"))]
         pub use __automatic_type_registration_impl::*;
     }
 }

--- a/crates/bevy_utils/src/debug_info.rs
+++ b/crates/bevy_utils/src/debug_info.rs
@@ -106,7 +106,7 @@ impl Deref for DebugName {
         #[cfg(feature = "debug")]
         return &self.name;
         #[cfg(not(feature = "debug"))]
-        return &FEATURE_DISABLED;
+        return FEATURE_DISABLED;
     }
 }
 

